doc/site/articles/technicalities-of-starting-a-match.markdown

```diff
@@ -0,0 +1,38 @@
+---
+layout: post
+title: Technicalities of starting a match
+parent: Articles
+permalink: articles/technicalities-of-starting-a-match
+author: sprunk
+---
+
+### Engine
+
+The only thing that the engine needs to start a game is a _start script_: plaintext data containing the setup such as game, map, players, modoptions, etc. Every other component's role is merely to facilitate the creation and exchange of startscripts.
+
+* in-engine Lua code can start a game via `Spring.Restart(string commandLineArgs, string startScript)`. This is available from LuaMenu (e.g. Chobby) but also regular ingame Lua instances.
+* external lobbies can run the engine and pass the startscript to standard input.
+* replays and save files generally work equivalently to a startscript (and contain the startscript for their game).
+* **practical tip**: during development it's convenient to have a `startscript.txt` file and drag-n-drop it to `spring.exe` (or equivalent). The engine leaves a `_script.txt` file containing the last used startscript so you can produce an intricate setup via a graphical lobby, or by watching a replay, and then reuse it.
+* **practical tip**: replays are associated with a specific engine version, so if you want to support watching old replays you'll need to handle it externally (it can be a simple script communicating with LuaMenu via a socket though, not necessarily a "full" lobby).
+
+### Lobby client
+
+Lobbies (either in-engine LuaMenu such as Chobby, or external programs) are generally responsible for obtaining a startscript to pass it to the engine.
+* for skirmish games, or multiplayer games where the player is the host, this means generating one locally.
+* for multiplayer it means receiving it from somewhere on the internet, alongside some metadata (such as "which of the listed players is me" or host IP). Usually it is received via the lobbyserver.
```

Comment:
One important bit here is that not a whole start script is needed, only the ip, port, user, pass. And those can be put directly on the command line too. That's IMHO quite a significant difference in the amount of information that needs to be handled.